### ISSUE TITLE
Friends with private profile are not displayed

### ISSUE THREAD

**Issue by kortemik** [jgeboski/bitlbee-steam#80]
Friends with private profile appear as offline even though client should be able to see their status as online and even the game they are playing.

**Comment by jgeboski** [comment]
Quite a while ago, the plugin began merging information on friends from two different sources.  The second source, the "state", is merged with the "summary," and overwrites any values which differ with the "state" value.  When the plugin first made this change, it was to address some games not appearing in the "summary."  Now it appears it is the other way around.  This issue is illustrated below with the JSON data, notice the persona states.

This is clearly an inconsistency on Steam's end, but the plugin can adjust to it since the correct information is received.  It would be really nice if one source would not only provide all of the information required, but valid information.

On a side note, this would also mean this bug exists within Steam's [Web Chat](https://steamcommunity.com/chat/).

Summary:
```
{
    "players": [
        {
            "steamid": "<snipped>",
            "communityvisibilitystate": 1,
            "profilestate": 1,
            "personaname": "<snipped>",
            "lastlogoff": 1429524940,
            "profileurl": "<snipped>",
            "avatar": "<snipped>",
            "avatarmedium": "<snipped>",
            "avatarfull": "<snipped>",
            "personastate": 1,
            "personastateflags": 0
        }
    ]
}
```

State:
```
{
    "m_unAccountID": <snipped>,
    "m_ulSteamID": "<snipped>",
    "m_strName": "<snipped>",
    "m_ePersonaState": 0,
    "m_nPersonaStateFlags": null,
    "m_strAvatarHash": "<snipped>",
    "m_tsLastMessage": 1429578597,
    "m_tsLastView": 1429578597
}
```

**Comment by kortemik** [comment]
Is there any possibility that we could suggest this to Valve, I mean it's most likely a design flaw which is not intended?

I think their developer portal might have a way to suggest things around and it might be a good way to show that we are doing this for good and they should not prevent us from using steam mobile api in future like many companies did, google gtalk for example.

**Comment by jgeboski** [comment]
I mean, it is possible, but not likely to do much.  It is just easier for the plugin to adjust.  Ideally, I would like to use [SteamKit](https://github.com/SteamRE/SteamKit) rather than the HTTP stuff anyways.

**Comment by jgeboski** [comment]
This should be resolved as of aaf2cbc.
